configure.ac

```diff
@@ -305,6 +306,42 @@ AC_CHECK_HEADER(valgrind/memcheck.h, [],
     AC_MSG_ERROR([valgrind requested but <valgrind/memcheck.h> not found.]))
 fi
 
+dnl Check for userfaultfd support
+have_userfaultfd=0
+AC_CHECK_HEADERS([linux/userfaultfd.h],
+    [AC_MSG_CHECKING([for userfaultfd interface])
+     AC_TRY_LINK([#include <sys/types.h>
+#include <linux/userfaultfd.h>
+#include <unistd.h>
+#include <sys/syscall.h>
+#include <fcntl.h>
+],
+        [return syscall(__NR_userfaultfd, O_CLOEXEC | O_NONBLOCK);],
+            [AC_MSG_RESULT([yes])
+             have_userfaultfd=1],
+            [AC_MSG_RESULT([no])
+             have_userfaultfd=0])])
+AS_IF([test "$have_userfaultfd" -eq "1"]
```

Comment:
`-eq` is a numeric comparison.  You should probably just have:

```
AS_IF([test $have_userfaultfd -eq 1],
```

(i.e., don't "case" the numbers into strings)

You're also missing a `,` at the end of this line.

Reply:
huh, how did that work...